### PR TITLE
feat: support argmin aten converter

### PR DESCRIPTION
**Pull request by bowang007**
# Description

Support argmin converter. 
#2498 

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Comment by bowang007** [comment]
Looks like we can deduplicate the code with `ArgMax` converter, we can also just leave it since the duplication is trivial here.

File: tests/py/dynamo/conversion/test_argmin_aten.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from .harness import DispatchTestCase
+
+
+class TestArgminConverter(DispatchTestCase):
+    @parameterized.expand(
+        [
+            # input dimension == 1
+            ("dim_1_keep_dim_true", (3,), 0, True),
+            ("dim_1_keep_dim_true", (3,), 0, False),
```
Comment (gs-olive) [inline tests/py/dynamo/conversion/test_argmin_aten.py:14]: Rename to `dim_1_keep_dim_false`

File: tests/py/dynamo/conversion/test_argmin_aten.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            # dim == None
+            ("dim_none", (3,), None, True),
+            ("dim_none", (3, 3), None, True),
+            ("dim_none", (3, 3, 3), None, False),
```
Comment (gs-olive) [inline tests/py/dynamo/conversion/test_argmin_aten.py:18]: Consider adding `_2d`, `_3d`, etc. to differentiate naming of test cases

**Review by gs-olive — COMMENTED** [review]
Added a few minor naming comments, otherwise looks good to me!

**Review by zewenli98 — COMMENTED** [review]
Consider combining `argmax.py` and `argmin.py` files (maybe could be moved to `reduce.py`?), and doing changes for their tests that @gs-olive mentioned above. Others LGTM!

**Comment by apbose** [comment]
The changes look good, but as Evan said, can `argmin.py` and `argmax.py` be merged? Also you could restructure it in such a way that it has a `base.py` in reduce folder exposing the topK layer in a function and argmin and argmax could use that. Else the code is redundant apart from the usage of the topK layer.

**Review by apbose — COMMENTED** [review]
The changes look good, but as Evan said, can argmin.py and argmax.py be merged? Also you could restructure it in such a way that it has a base.py in reduce folder exposing the topK layer in a function and argmin and argmax could use that. Else the code is redundant apart from the usage of the topK layer.

**Review by gs-olive — COMMENTED** [review]
Overall looks good - the new folder `argmaxmin` just needs to be added to [`setup.py`](https://github.com/pytorch/TensorRT/blob/2f9b2594ea3599fa59b69cd69d4d87b323d97f59/setup.py) to reflect in the build

**Comment by zewenli98** [comment]
@bowang007 As we discussed offline, since `argmin` and `argmax` use `ITopKLayer`, for generality, we can have a file `impl/topk.py` and move related converters into this file.

**Comment by bowang007** [comment]
> Overall looks good - the new folder `argmaxmin` just needs to be added to [`setup.py`](https://github.com/pytorch/TensorRT/blob/2f9b2594ea3599fa59b69cd69d4d87b323d97f59/setup.py) to reflect in the build

Thanks @gs-olive !
I refactored that folder into a single file `topk.py`. 
Btw, I'm not so familiar with python packages but does this mean every time we create a folder we always need to build the package for that folder right? 
Thanks!

**Comment by bowang007** [comment]
> @bowang007 As we discussed offline, since `argmin` and `argmax` use `ITopKLayer`, for generality, we can have a file `impl/topk.py` and move related converters into this file.

@zewenli98 Thanks!
I refactored that part, should be good to go now. 
Please take a look so we can merge it by EOD.

**Review by zewenli98 — APPROVED** [review]
Looks good to me!

**Review by gs-olive — APPROVED** [review]
Looks good, pending CI

**Pull request merged**